deploy-agent/deployd/common/utils.py

```diff
@@ -207,7 +207,7 @@ def get_info_from_facter(keys) -> Optional[dict]:
         # increment stats - facter calls
         create_sc_increment('deployd.stats.internal.facter_calls_sum', 1)
         log.info(f"Fetching {keys} keys from facter")
-        cmd = ['facter', '-jp']
+        cmd = ['AWS_EC2_METADATA_V1_DISABLED=true', 'facter', '-jp']
```

Comment:
should it be set in cronjob in puppet? https://sourcegraph.pinadmin.com/github.com/pinternal/puppet-control-repo/-/blob/modules/teletraan/manifests/cron.pp?L10:125

Reply:
If we configure this within a Puppet cron job, we'll need to wait for the Puppet deployment to complete; similarly, if we set the environment variable in this Pull Request as proposed, we'll have to wait for the new deployd version to be fully deployed. Therefore, it is analogous in both scenarios.

Reply:
discussed off line